Mobile_Detect.php

```diff
@@ -1147,22 +1131,35 @@ public function isMobile($userAgent = null, $httpHeaders = null)
             $this->setUserAgent($userAgent);
         }
 
-        // Check specifically for cloudfront headers if the useragent === 'Amazon CloudFront'
-        if ($this->getUserAgent() === 'Amazon CloudFront') {
-            $cfHeaders = $this->getCfHeaders();
-            if(array_key_exists('HTTP_CLOUDFRONT_IS_MOBILE_VIEWER', $cfHeaders) && $cfHeaders['HTTP_CLOUDFRONT_IS_MOBILE_VIEWER'] === 'true') {
-                return true;
-            }
+        if ($this->isViaAmazonCloudFront('mobile')) {
+            return true;
         }
 
         $this->setDetectionType(self::DETECTION_TYPE_MOBILE);
 
         if ($this->checkHttpHeadersForMobile()) {
             return true;
-        } else {
-            return $this->matchDetectionRulesAgainstUA();
         }
 
+        return $this->matchDetectionRulesAgainstUA();
+    }
+
+    /**
+     * Decides whether user agent matches Amazon CloudFront and request is via mobile or tablet
+     *
+     * @param type string mobile|tablet
+     * @return boolean
+     */
+    public function isViaAmazonCloudFront($type)
```

Comment:
I think this is a good abstraction. Can you change your PR to reflect **only** this change?
Leave the rest of optimizations in a separate PR so I can review them separately. Optimizations are easier to review when they are isolated.